cpp/source/deal.II/geometry.h

```diff
@@ -72,6 +81,13 @@ class Geometry
     return _materials;
   }
 
+  void set_materials(
+      std::shared_ptr<std::unordered_map<
+          std::string, std::vector<dealii::types::material_id>>> materials)
+  {
+    _materials = materials;
+  }
+
```

Comment:
Would it make sense to have a ``std::unordered_map<std::string, std::vector<dealii::types::boundary_id>>> _boundaries`` (or a ``std::shared_ptr<...>``) in ``Geometry<dim>``?